modules/aws/sonar-base-instance/secret.tf

```diff
@@ -16,38 +16,56 @@
 # https://registry.terraform.io/providers/hashicorp/tls/latest/docs/resources/private_key
 ###################################################################################
 
+data "aws_secretsmanager_secret" "sonarw_private_key_secret_data" {
+  count = var.internal_private_key_secret_name != null ? 1 : 0
+  name = var.internal_private_key_secret_name
+}
+
+data "aws_secretsmanager_secret" "password_secret_data" {
+  count = var.web_console_admin_password_secret_name != null ? 1 : 0
+  name = var.web_console_admin_password_secret_name
+}
+
 resource "tls_private_key" "sonarw_private_key" {
+  count     = var.internal_private_key_secret_name == null ? 1 : 0
   algorithm = "RSA"
   rsa_bits  = 4096
 }
 
 locals {
-  primary_node_sonarw_public_key  = !var.hadr_secondary_node ? "${chomp(tls_private_key.sonarw_private_key.public_key_openssh)} produced-by-terraform" : var.sonarw_public_key
-  primary_node_sonarw_private_key = !var.hadr_secondary_node ? chomp(tls_private_key.sonarw_private_key.private_key_pem) : var.sonarw_private_key
-  sonarw_secret_aws_arn           = aws_secretsmanager_secret.sonarw_private_key_secret.arn
-  sonarw_secret_aws_name          = aws_secretsmanager_secret.sonarw_private_key_secret.name
+  primary_node_sonarw_public_key  = var.internal_public_key != null ? var.internal_public_key : (!var.hadr_secondary_node ? "${chomp(tls_private_key.sonarw_private_key[0].public_key_openssh)} produced-by-terraform" : var.sonarw_public_key)
+  primary_node_sonarw_private_key = var.internal_private_key_secret_name != null ? var.internal_private_key_secret_name : (!var.hadr_secondary_node ? chomp(tls_private_key.sonarw_private_key[0].private_key_pem) : var.sonarw_private_key)
+  sonarw_secret_aws_arn           = var.internal_private_key_secret_name == null ? aws_secretsmanager_secret.sonarw_private_key_secret[0].arn : data.aws_secretsmanager_secret.sonarw_private_key_secret_data[0].arn
+  sonarw_secret_aws_name          = var.internal_private_key_secret_name == null ? aws_secretsmanager_secret.sonarw_private_key_secret[0].name : data.aws_secretsmanager_secret.sonarw_private_key_secret_data[0].name
+
+  password_secret_aws_arn = var.web_console_admin_password_secret_name == null ? aws_secretsmanager_secret.password_secret[0].arn : data.aws_secretsmanager_secret.password_secret_data[0].arn
+  password_secret_name = var.web_console_admin_password_secret_name == null ? aws_secretsmanager_secret.password_secret[0].name : var.web_console_admin_password_secret_name
 
-  password_secret_aws_arn = aws_secretsmanager_secret.password_secret.arn
+  should_create_password_in_secret_manager = var.web_console_admin_password_secret_name == null ? true : false
```

Comment:
should_create_password_in_secrets_manager = var.web_console_admin_password_secret_name == null. Also, secrets in plural. Also this creates asymmetry between the two secrets - we should also add: should_create_sonarw_private_key_in_secrets_manager